### PR TITLE
e2e: reduce defaultCloneCount to 3

### PR DESCRIPTION
**Pull request by pkalever**
# Describe what this PR does #

CI is failing very frequently hitting resource leaks issue,
until we solve the root cause for resource leaks reducing the clone
count from 10 to 3.

updates #2327 

Signed-off-by: Prasanna Kumar Kalever \<prasanna.kalever@redhat.com\>

### CODE REVIEW

File: e2e/rbd.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ var (
 	snapshotPath           = rbdExamplePath + "snapshot.yaml"
 	deployFSAppPath        = e2eTemplatesPath + "rbd-fs-deployment.yaml"
 	deployBlockAppPath     = e2eTemplatesPath + "rbd-block-deployment.yaml"
-	defaultCloneCount      = 10
+	defaultCloneCount      = 3 // TODO: set to 10
```
Comment (Rakshith-R) [inline e2e/rbd.go]: Please link the issue #2327 and add a comment why it is being done.

File: e2e/rbd.go [hunk 2]
```diff
@@ -187,7 +187,7 @@ func validateRBDImageCount(f *framework.Framework, count int, pool string) {
 		e2elog.Failf("failed to list rbd images: %v", err)
 	}
 	if len(imageList) != count {
-		e2elog.Failf(
+		e2elog.Logf(
```
Comment (Rakshith-R) [inline e2e/rbd.go]: We decided to keep this as Failf ? https://cephcsi.slack.com/archives/C025R9CNS3H/p1656479466791209?thread_ts=1656478083.987689&cid=C025R9CNS3H

**Review by Rakshith-R — CHANGES_REQUESTED** [review]

File: e2e/rbd.go (hunk 2 above)
Comment (pkalever) [reply e2e/rbd.go]: accidental change.

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM , lets see how CI goes

**Comment by pkalever** [comment]
/retest ci/centos/mini-e2e-helm/k8s-1.22
[logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e-helm_k8s-1.22/runs/3652/nodes/103/log/?start=0)

**Comment by pkalever** [comment]
@Madhu-1 @Rakshith-R can you provide your acks, the e2e runs are much better than before.

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Pull request merged**